homeassistant/components/azure_devops/sensor.py

```diff
@@ -39,19 +39,23 @@ async def async_setup_entry(
         AzureDevOpsSensor(
             coordinator,
             AzureDevOpsSensorEntityDescription(
-                key=f"{build.project.id}_{build.definition.id}_latest_build",
+                key=f"{build.project.project_id}_{build.definition.build_id}_latest_build",
                 translation_key="latest_build",
                 translation_placeholders={"definition_name": build.definition.name},
                 attrs=lambda build: {
-                    "definition_id": build.definition.id,
-                    "definition_name": build.definition.name,
-                    "id": build.id,
+                    "definition_id": build.definition.build_id
+                    if build.definition
+                    else None,
+                    "definition_name": build.definition.name
+                    if build.definition
+                    else None,
+                    "id": build.build_id,
```

Comment:
Please avoid multi-line ternary operators, you could probably use parentheses here.